platformdirs.py

```diff
@@ -36,6 +36,8 @@
         # are actually checked for and the rest of the module expects
         # *sys.platform* style strings.
         system = 'linux2'
+elif os.getenv('EXTERNAL_STORAGE'):
```

Comment:
Is this as specific as we can be?

Reply:
yep

Reply:
There has to be a direct way to detect this, this would be super brittle.

There's got to be people on non-Android devices out there even who have this environment variable set to some NAS of theirs :D

Reply:
More helpfully perhaps -- `uname -a` shows Android for me on my device, so I strongly suspect there's even something in `/proc` that contains this info no?

Reply:
I was guided by [kivy](https://github.com/kivy/kivy/blob/master/kivy/utils.py#L433) and it seems that this is a normal practice

Reply:
`/proc` does not contains any helpful information, because all files not readable from any app (not including catalogs processes created by the application itself). Yes, i have already tried `uname -a` and I didn't find anything interesting either
```
Linux localhost 3.18.140-19180804 #1 SMP PREEMPT Thu May 6 15:00:11 KST 2021 armv8l
```

Reply:
I don't see that environment variable referenced in the kivy code you linked. Am I missing it?

Reply:
And skimming it mentally I actually don't see how it'd return `android` on anything but python-for-android -- but I don't have it installed anywhere to double check -- that kivy function works for you on your device?

Reply:
Nope, i just want say, that using env variables to define platform is normal

Reply:
Oh, well of course :), but I'm commenting on the specific one. I don't think doing it this way is safe, that environment variable can be used on other operating systems, and where those other operating systems aren't doing something weird. Whereas the kivy ones are specifically android-specific environment variables, and someone setting it on a linux machine knows they're doing something weird.

Reply:
Well, I can suggest making this a little more specific (take the android environment constant and compare it), e.g.:
```python
os.getenv('ANDROID_DATA') == '/data'
```
Also, for greater accuracy, you can add more constants: ANDROID_STORAGE == '/storage', ANDROID_ROOT == '/system', etc

Reply:
Yeah, any of those definitely sound much safer to me -- do some combination of them work across devices?

Reply:
Yes, it should work across devices. I have a lot of android devices, and before merging, I'll test it properly